### ISSUE TITLE
failed to list *v1.ConfigMap: configmaps is forbidden

### ISSUE THREAD

**Issue by g-bohncke** [aws/karpenter-provider-aws#6975]
### Description

**Observed Behavior**:
see issue : https://github.com/aws/karpenter-provider-aws/issues/5383
**Expected Behavior**:
That karpenter works after an upgrade
**Reproduction Steps** (Please include YAML):
Upgrade from 0.37.2 -> 0.37.3

**Versions**:
- Chart Version:
-  0.37.3

Looks like this old bug is back in the latest release

**Comment by jonathan-innis** [comment]
> Looks like this old bug is back in the latest release

There shouldn't have been any change here beyond the addition of the conversion webhooks being on by default. Can you share the full log dump here that you are running into?

**Comment by jfoechsler** [comment]
I hit the same when trying to disable webhooks
Role has this:
```
{{- if .Values.webhook.enabled }}
  - apiGroups: [""]
    resources: ["configmaps", "secrets"]
    verbs: ["get", "list", "watch"]
{{- end }}
```

**Comment by g-bohncke** [comment]
Yes it the same for me I want those webhooks disabled because I want to control the update via argocd 
I have them both on the same value of false and that doesn't work it looks like what [jfoechsler](https://github.com/jfoechsler) mentioned that role binding has to eager if statement.

**Comment by dcherniv** [comment]
Same problem. Disabling webhooks on 0.37.3 breaks karpenter completely.

**Comment by jonathan-innis** [comment]
@g-bohncke @jfoechsler @dcherniv @mhobbs812 Are you all hitting this issue when you have webhooks disabled on v0.37.3?

**Comment by dcherniv** [comment]
Yes. 0.37.3 with webhooks disabled fails with this error. 0.37.2 works fine with webhooks disabled.
